### PR TITLE
Use a dispatch system for sockets

### PR DESCRIPTION
**Pull request by Zarel**
Instead of using a mega-switch, instead use a dispatch system for the sockets protocol.

A big advantage is that we don't need to be weird about variable declarations using this method. We can use `const`s!

### CODE REVIEW

**Comment by Zarel** [comment]
What do you all think about migrating the rest of our mega-switches to use this method?

**Comment by mia-pi-git** [comment]
Personally, I'd favor that as well - cleaner, and easier to modify.

File: server/sockets.ts [hunk 1]
```diff
@@ -408,137 +526,9 @@ export class ServerStream extends Streams.ObjectReadWriteStream<string> {
...
 	_write(data: string) {
 		// console.log('worker received: ' + data);
-		let socket: import('sockjs').Connection | undefined = undefined;
-		let socketid = '';
-		let room: Map<string, import('sockjs').Connection> | undefined = undefined;
-		let roomid = '' as RoomID;
-		let roomChannel: Map<string, ChannelID> | undefined = undefined;
-		let channelid: ChannelID = 0;
-		let nlLoc = -1;
-		let message = '';
```
Comment (Zarel) [inline server/sockets.ts:418]: These are gone forever! Always the most awkward parts of mega-switches. I've experimented with several approaches in the past and they've all sucked.

**Comment by Zarel** [comment]
This won't take effect until next restart so I'm not in a huge rush to merge.

**Pull request merged**